lib/fake_data.rb

```diff
@@ -0,0 +1,115 @@
+require "csv"
+
+class FakeData
+  def self.insert
+    new.insert
+  end
+
+  def self.delete
+    new.delete
+  end
+
+  def insert
+    if has_test_subscribers?
+      puts "There is already test data in the system."
+      puts "Run rake fake_data:delete first."
+      return
+    end
+
+    fake_subscriptions_data.each do |subscription_stat|
+      subscriber_ids = create_subscribers(subscription_stat.number_of_subscribers).ids
+      create_subscriptions(subscriber_ids, subscription_stat.number_of_subscriptions)
+    end
+
+    count = existing_test_subscribers.count
+    puts "Created #{count} subscribers."
+  end
+
+  def delete
+    count = existing_test_subscribers.delete_all
+    puts "Deleted #{count} subscribers."
+  end
+
+private
+
+  def frequencies
+    @frequencies ||= begin
+      Enumerator.new do |yielder|
+        loop do
+          3.times { yielder << :weekly }
+          7.times { yielder << :daily }
+          10.times { yielder << :immediately }
+        end
+      end
+    end
+  end
+
+  def subscriber_list_ids
+    @subscriber_list_ids ||= begin
+      Enumerator.new do |yielder|
+        loop do
+          SubscriberList.order("RANDOM()").pluck(:id).each do |id|
```

Comment:
Should we grab all of this once and then pick a random one (no idea if this is part of performance issues mind)

Reply:
Not a performance issue, but it makes sense, I've added that in 40a2b5b.